compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirDeclarationCheckerUtils.kt

```diff
@@ -38,16 +38,14 @@ internal fun checkExpectDeclarationVisibilityAndBody(
     }
 }
 
-internal fun checkPropertyInitializer(
+internal fun checkProperty(
```

Comment:
Perhaps this is quite personal/cosmetic issue, but I still prefer to separate stuff in some logical chunk, like checkPropertyInitializer, checkPropertyAccessors, etc., then you can still use this, checkProperty, as a one entry point.

Reply:
I considered that but there's only one extra check on the accessor, and the current function also does some examination of the accessors too so it's not clear what each function would be responsible for.

Reply:
> the current function also does some examination of the accessors too

If you refer to `hasAccessorImplementation`, it's used to differentiate diagnostics for uninitialized properties. So, in terms of reported diagnostics, all were related to property initializer.

But, again, this is cosmetic stuff. When we have more diagnostics for accessors, we can revisit those and make them as a separate util like I suggested.

Reply:
I'm for splitting that function too. We can at least extract the new check to `fun checkForValpropertyWithSetter(...)`

Reply:
I split up the function into `checkPropertyInitializer` (original) and `checkPropertyAccessors` (so we don't have to rename it again in the future if the name was too specific)

Reply:
I'm even removing this at https://github.com/JetBrains/kotlin/pull/4178 as per https://github.com/JetBrains/kotlin/pull/4142#discussion_r587280893.